pkg/api/v1alpha1/cluster_types.go

```diff
@@ -372,6 +372,19 @@ type ClusterStatus struct {
 	// Descriptive message about a fatal problem while reconciling a cluster
 	// +optional
 	FailureMessage *string `json:"failureMessage,omitempty"`
+	// EksdReleaseRef defines the properties of the EKS-D object on the cluster
+	EksdReleaseRef EksdReleaseRef `json:"eksdReleaseRef,omitempty"`
+}
+
+type EksdReleaseRef struct {
+	// ApiGroup refers to the EKS-D API grouo
+	ApiGroup string `json:"apiGroup,omitempty"`
+	// Kind refers to the Release kind for the EKS-D object
+	Kind string `json:"kind,omitempty"`
+	// Name refers to the name of the EKS-D object on the cluster
+	Name string `json:"name,omitempty"`
+	// Namespace refers to the namespace for the EKS-D release resources
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
I would vote to make all these fields inside `EksdReleaseRef` required
```suggestion
	ApiGroup string `json:"apiGroup"`
	// Kind refers to the Release kind for the EKS-D object
	Kind string `json:"kind"`
	// Name refers to the name of the EKS-D object on the cluster
	Name string `json:"name"`
	// Namespace refers to the namespace for the EKS-D release resources
	Namespace string `json:"namespace"`
```

Reply:
Updated! Right now the fields are empty in those test files but it will be populated when I implement the logic in the create/upgrade workflow in the next PR